<a href="https://colab.research.google.com/github/Yurakc/platzi_data/blob/main/Aprendizaje%20Autom%C3%A1tico%20y%20Miner%C3%ADa%20De%20Datos/sesion%203/2_Estadisticos_descriptivos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import scipy.stats as stats

In [ ]:
pesos = [100, 150, 150, 200, 250, 300, 325, 400,415, 500, 600, 1000]

In [ ]:
pesos

[100, 150, 150, 200, 250, 300, 325, 400, 415, 500, 600, 1000]

In [ ]:
len(pesos)

12

In [ ]:
pesos[-2]

600

# Media

In [ ]:
pesos_media = np.mean(pesos)
pesos_media

365.8333333333333

# Mediana

In [ ]:
np.median(pesos)

312.5

# Moda

In [ ]:
stats.mode(pesos)

ModeResult(mode=array([150]), count=array([2]))

# Rango

In [ ]:
np.max(pesos) - np.min(pesos)

900

# Cuartiles y IQR

In [ ]:
stats.mstats.mquantiles(pesos)

array([172.5 , 312.5 , 461.75])

In [ ]:
pesos_cuartiles = pd.qcut(pesos, 4)

pesos_cuartiles.categories

IntervalIndex([(99.999, 187.5], (187.5, 312.5], (312.5, 436.25], (436.25, 1000.0]],
              closed='right',
              dtype='interval[float64]')

La funcion pandas.qcut convierte una serie en una lista de N cuantiles. Si le pasamos 4, tenemos los cuartiles. qcut devuelve una serie categórica nueva con los cuantiles como categorias

In [ ]:
pesos_cuartiles = pd.qcut(pesos, 4)

pesos_cuartiles.categories

IntervalIndex([(99.999, 187.5], (187.5, 312.5], (312.5, 436.25], (436.25, 1000.0]],
              closed='right',
              dtype='interval[float64]')

Aqui se ve que los intervalos que separan los pesos en 4 bloques son:

Categories (4, interval[float64]): [(99.999, 187.5] < (187.5, 312.5] < (312.5, 436.25] < (436.25, 1000.0]]

con 187.5 siendo Q1 y 436.25 siendo Q3

Es importante notar que hay diversas formas de obtener los cuartiles, de aqui la diferencia con el método calculado a mano

In [ ]:
IQR = 436.25 - 187.5
IQR

248.75

# Desviación Estándar

In [ ]:
pesos_std = np.std(pesos)
pesos_std

239.6771555423856